tests/badness_detection.go

```diff
@@ -48,13 +58,30 @@ func TestBadness(t *testing.T, hashers ProtoHashers) {
 		// Extensions are bad.
 		&pb2_latest.BadWithExtensions{},
 
+		// Write some data to the extension field.
+		func() proto.Message {
+			// Use an unregistered extension to keep each test simple and self-contained.
+			var E_SpecialNumber = &proto.ExtensionDesc{
+				ExtendedType:  (*pb2_latest.BadWithExtensions)(nil),
+				ExtensionType: (*int32)(nil),
+				Field:         123,
+				Name:          "schema.proto2.special_number",
+				Tag:           "varint,123,opt,name=special_number,json=specialNumber",
+				Filename:      "bad.proto",
+			}
+
+			v := &pb2_latest.BadWithExtensions{Text: proto.String("Test")}
+			proto.SetExtension(v, E_SpecialNumber, 42)
+			return v
+		}(),
+
 		// Create proto messages with unknown fields. That's bad.
 		forgetAllFields(&pb2_latest.PersonV1{Name: proto.String("Unbekannt")}),
 	}
 	for _, message := range badProtos {
 		_, err := hasher.HashProto(message)
 		if err == nil {
-			t.Errorf("Attempting to hash %T{ %+v} should have returned an error.", message, message)
+			t.Errorf("Attempting to hash %T{ %[1]v } should have returned an error.", message)
```

Comment:
Any optimization that this would afford complicates the simple straight-forward construction that was there before, and is unnecessary, as it is, itself, a test. And thus not performance critical.

Reply:
This was not changed to optimize for performance, but rather for readability.

Reply:
You also drop the “+” flag, so this isn’t just a readability change. And no, it does not really meet the bar of “better readability”… they are equivalently readable, and the use of indexed references is so uncommon, that it can be argued that using it where reduplication works just as well would not be increased readability.

I mean, whom are you targeting here for readability? Pedantic people who know all the language features, or the general programmer?

Reply:
Dropping the "+" flag was a separate change made because it makes no difference when applied to protos. The output is the same before and after.

I agree that it's not a very common feature but it is straightforward enough that I don't think it would be confusing for someone who is familiar with other flags and encounters it for the first time.